osu.Game/Online/Chat/MessageNotifier.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Humanizer;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Logging;
+using osu.Game.Configuration;
+using osu.Game.Graphics;
+using osu.Game.Online.API;
+using osu.Game.Overlays;
+using osu.Game.Overlays.Notifications;
+using osu.Game.Users;
+
+namespace osu.Game.Online.Chat
+{
+    /// <summary>
+    /// Component that handles creating and posting notifications for incoming messages.
+    /// </summary>
+    public class MessageNotifier : Component
+    {
+        [Resolved(CanBeNull = true)]
+        private NotificationOverlay notificationOverlay { get; set; }
+
+        [Resolved(CanBeNull = true)]
+        private ChatOverlay chatOverlay { get; set; }
+
+        [Resolved(CanBeNull = true)]
+        private ChannelManager channelManager { get; set; }
+
+        private Bindable<bool> notifyOnMention;
+        private Bindable<bool> notifyOnChat;
+        private Bindable<User> localUser;
+
+        /// <summary>
+        /// Determines if the user is able to see incoming messages.
+        /// </summary>
+        public bool IsActive => chatOverlay?.IsPresent == true;
```

Comment:
bad name, also shouldn't be public. also is only used in one place so it might be fine to just inline it.

also the condition you want is probably `chatOverlay?.State == Visible`